mimic/test/test_identity_auth.py

```diff
@@ -826,6 +826,53 @@ def test_response_for_validate_token(self):
         self.assertTrue(len(json_body['access']['user']['roles']) > 0)
         self.assertTrue(json_body['access'].get('serviceCatalog') is None)
 
+    def test_response_for_validate_token_from_another_account(self):
+        """
+        Test to verify :func: `validate_token` when tenant_id is not
+        provided using the argument `belongsTo` and a different token
+        is provided via the `x-auth-token` header.
+
+        Note: This is how authentication validators like Repose operate.
+        """
+        core, root = core_and_root([ExampleAPI()])
+
+        user = {
+            "username": "user123456",
+            "password": "654321resu"
+        }
+        admin = {
+            "username": "admin",
+            "password": "nimda"
+        }
+
+        (user_response, user_data) = authenticate_with_username_password(
+            self, root, username=user["username"], password=user["password"])
+
+        (admin_response, admin_data) = authenticate_with_username_password(
+            self, root, username=admin["username"], password=admin["password"])
+
+        user_token = user_data['access']['token']['id']
+        user_tenant = user_data['access']['token']['tenant']['id']
+
+        admin_token = admin_data['access']['token']['id']
+        admin_tenant = admin_data['access']['token']['tenant']['id']
```

Comment:
Minor stylistic nitpick: given that you're repeating the same 4 lines for user and admin, maybe refactoring into a short utility function could make the test shorter?  This could also eliminate the variables `user` and `admin`, leaving you with osmething like:

```python
user_token, user_tenant = tenant_and_token("user123456", "654321resu")
admin_token, admin_tenant = tenant_and_token("admin", "nimda")
```
which could take 20 lines (including blanks) down to 2; seems like a worthwhile refactoring :).

The main reason I propose this is that the more bound names there are in a particular scope, the more carefully one must read it, and the more stuff you have to remember as you work your way down.  It's not the pure line count that I'm concerned about so much as the fact that as I'm reading this test I have to keep asking myself "is `user_response` used again?  do I need to know about `admin_data`?"